nav2_lifecycle_manager/src/lifecycle_manager.cpp

```diff
@@ -248,9 +256,95 @@ LifecycleManager::resume()
   return true;
 }
 
-// TODO(mjeronimo): This is used to emphasize the major events during system bring-up and
-// shutdown so that the messgaes can be easily seen among the log output. We should replace
-// this with a ROS2-supported way of highlighting console output, if possible.
+void
+LifecycleManager::createBondTimer()
+{
+  if (bond_timeout_.count() <= 0) {
+    return;
+  }
+
+  message("Creating bond timer...");
+
+  bond_timer_ = this->create_wall_timer(
+    100ms,
+    std::bind(&LifecycleManager::checkBondConnections, this));
+}
+
+void
+LifecycleManager::destroyBondConnections()
+{
+  bond_timer_.reset();
+
+  if (bond_map_.empty()) {
+    return;
+  }
+
+  message("Terminating bond connections...");
+
+  for (auto & node_name : node_names_) {
+    bond_map_[node_name]->breakBond();
+  }
+
+  bond_map_.clear();
+}
+
+void
+LifecycleManager::createBondConnections()
+{
+  const double timeout_ns =
+    std::chrono::duration_cast<std::chrono::nanoseconds>(bond_timeout_).count();
+  const double timeout_s = timeout_ns / 1e9;
+
+  for (auto & node_name : node_names_) {
+    bond_map_[node_name] =
+      std::make_shared<bond::Bond>("bond", node_name, service_client_node_);
```

Comment:
There is a difference in the ** bond id** between lifecycle_manger (where `node_name` is being used) and the lifecycle_node (where `controller_server` is being used). Doesn't the id need to be the same between the lifecycle_manger and a lifecycle_node, and unique for each lifecycle_node?

https://github.com/ros-planning/navigation2/blob/227c30ad4c84e839a6aff9fc6cfcb95ceabb833e/nav2_util/src/lifecycle_node.cpp#L83-L85

Reply:
:-) You found my quirk, that is correct. Copy pasta error from my initial prototypes using the controller server as a demo.